### ISSUE TITLE
VM backup fleecing doesn't work with storage=drbd

### ISSUE THREAD

**Issue by fdlab76** [LINBIT/linstor-proxmox#63]
When I use the "fleecing" feature for backup I get the following error

INFO: starting new backup job: vzdump 100 --storage QNAP --compress zstd --fleecing '1,storage=drbd_0' --notes-template '{{guestname}}' --mode snapshot --node pve-2 --mailnotification always --all 0 --mailto 
INFO: Starting Backup of VM 100 (qemu)
INFO: Backup started at 2024-07-12 22:05:40
INFO: status = running
INFO: VM Name: Debian
INFO: include disk 'scsi0' 'drbd_1:pm-098df180_100' 33555416K
INFO: backup mode: snapshot
INFO: ionice priority: 7
INFO: creating vzdump archive '/mnt/pve/QNAP/dump/vzdump-qemu-100-2024_07_12-22_05_40.vma.zst'
ERROR: allocated name ('vm-100-fleece-0') has to be a valid UUID, legacy, state, or cloud-init name at /usr/share/perl5/PVE/Storage/Custom/LINSTORPlugin.pm line 385.
INFO: aborting backup job
INFO: resuming VM again
ERROR: Backup of VM 100 failed - allocated name ('vm-100-fleece-0') has to be a valid UUID, legacy, state, or cloud-init name at /usr/share/perl5/PVE/Storage/Custom/LINSTORPlugin.pm line 385.
INFO: Failed at 2024-07-12 22:05:40
INFO: Backup job finished with errors
INFO: notified via target 
TASK ERROR: job errors

Configuration:

Proxmox 8.2.4

linstor-client                       1.23.0-1 
linstor-common                 1.28.0-1  
linstor-controller                1.28.0-1 
linstor-proxmox                  8.0.3-1       
linstor-satellite                    1.28.0-1     
python-linstor                    1.23.0-1

**Comment by rck** [comment]
thanks for reporting, the fix should be trivial, but whatever I try I can not find any checkbox where I could enable this "fleecing" in the GUI. you have a VM, obviously on DRBD/LINSTOR, and now you click on the "backup" tab of the VM, and then? I can select my "local" (dir) and a "local-zfs" (also "dir" that I created) as backup storage but there is no checkbox for fleecing or anything?

This GUI always confuses the hell out of me... can you give some more information, screenshots welcome. Or did you run the job on the CLI?

**Comment by rck** [comment]
did you trigger the backup job from the GUI? where can I find the "fleecing" checkbox? I assume this is a VM on DRBD. and then you click on the "backup" tab for that VM and then? what kind of storage do you select and where would the "fleecing" checkbox/option be? screenshots welcome

**Comment by rck** [comment]
okay, found it, it had to be a "backup job" in the "datacenter", not a direct backup of a VM  :woozy_face:

**Comment by rck** [comment]
can you confirm that 264b1b71e50f4db6d6ff1af7ef05c7f6698e2b3b fixes the issue? getting the checkout and doing something like 

```
#!/usr/bin/env bash
set -eu

[ $# -eq 1 ] || exit 1
HOST=$1

scp ./LINSTORPlugin.pm root@"${HOST}":/usr/share/perl5/PVE/Storage/Custom/
scp DRBDPlugin.pm.divert root@"${HOST}":/usr/share/perl5/PVE/Storage/DRBDPlugin.pm
ssh root@"${HOST}" mkdir /usr/share/perl5/LINBIT 2>/dev/null || true
scp ./LINBIT/Linstor.pm root@"${HOST}":/usr/share/perl5/LINBIT/Linstor.pm
scp ./LINBIT/PluginHelper.pm root@"${HOST}":/usr/share/perl5/LINBIT/PluginHelper.pm
ssh root@"${HOST}" "systemctl restart pvedaemon"
echo "OK: ${HOST}"
```

should do

**Comment by fdlab76** [comment]
Great job, works perfectly.

Thank you very much.

backup log below

==NFO: starting new backup job: vzdump 100 --all 0 --node pve-2 --mailto mail@example.net --notes-template '{{guestname}}' --mode snapshot --storage QNAP --compress zstd --fleecing '1,storage=drbd_0' --mailnotification failure
INFO: Starting Backup of VM 100 (qemu)
INFO: Backup started at 2024-07-15 12:47:12
INFO: status = running
INFO: VM Name: Debian
INFO: include disk 'scsi0' 'drbd_1:pm-098df180_100' 33555416K
INFO: exclude disk 'scsi1' 'drbd_1:vm-100-disk-2' (backup=no)
INFO: exclude disk 'scsi2' 'drbd_0:pm-bc57a65d_100' (backup=no)
INFO: backup mode: snapshot
INFO: ionice priority: 7
INFO: creating vzdump archive '/mnt/pve/QNAP/dump/vzdump-qemu-100-2024_07_15-12_47_12.vma.zst'

NOTICE
  Trying to create diskful resource (vm-100-fleece-0) on (pve-2).
INFO: drive-scsi0: attaching fleecing image drbd_0:vm-100-fleece-0 to QEMU
INFO: issuing guest-agent 'fs-freeze' command
INFO: issuing guest-agent 'fs-thaw' command
INFO: started backup task 'd2772034-e08a-47bf-8461-54547856d568'
INFO: resuming VM again
INFO:   2% (913.5 MiB of 32.0 GiB) in 3s, read: 304.5 MiB/s, write: 240.6 MiB/s
INFO:   6% (2.2 GiB of 32.0 GiB) in 6s, read: 442.2 MiB/s, write: 190.0 MiB/s
INFO:  23% (7.5 GiB of 32.0 GiB) in 9s, read: 1.8 GiB/s, write: 19.4 MiB/s
INFO:  31% (10.1 GiB of 32.0 GiB) in 12s, read: 870.7 MiB/s, write: 135.0 MiB/s
INFO:  41% (13.3 GiB of 32.0 GiB) in 15s, read: 1.1 GiB/s, write: 32.1 MiB/s
INFO:  56% (18.1 GiB of 32.0 GiB) in 18s, read: 1.6 GiB/s, write: 138.8 MiB/s
INFO:  65% (20.9 GiB of 32.0 GiB) in 21s, read: 959.4 MiB/s, write: 76.3 MiB/s
INFO:  78% (25.2 GiB of 32.0 GiB) in 24s, read: 1.4 GiB/s, write: 60.2 MiB/s
INFO:  97% (31.2 GiB of 32.0 GiB) in 27s, read: 2.0 GiB/s, write: 57.7 MiB/s
INFO: 100% (32.0 GiB of 32.0 GiB) in 29s, read: 398.5 MiB/s, write: 342.0 KiB/s
INFO: backup is sparse: 29.22 GiB (91%) total zero data
INFO: transferred 32.00 GiB in 29 seconds (1.1 GiB/s)
INFO: archive file size: 781MB
INFO: adding notes to backup
INFO: prune older backups with retention: keep-daily=15
INFO: pruned 0 backup(s)
INFO: Finished Backup of VM 100 (00:00:55)
INFO: Backup finished at 2024-07-15 12:48:07
INFO: Backup job finished successfully
TASK OK

**Comment by rck** [comment]
thanks for testing. some other things accumulated, so I guess I will do a release very soon anyways. just one thing: DRBD should support that use case, sure, but I'm really not sure if it is a good idea to use a DRBD disk for that in the first place? according to my understanding this fleece-disk is like a scratchpad, and it should be local and fast. I'd guess other types would make more sense in this particular case (e.g., local lvmthin). VM data on DRBD for obvious reasons, fleece scratch space on local disk space. no?

**Comment by fdlab76** [comment]
Of course, you are completely right, but when I configured the cluster the fleecing option did not exist and I allocated all the space as DRDB and I need to resize it to create lvm-thin local storage...
I need fleecing for backup on a really slow NAS and without it I have problems saving a VM with a database that writes continuously.
I will definitely do this during a cluster maintenance interval.
Thanks again for the support.
Kind regards
